### ISSUE TITLE
Yarn help changes text color in powershell

### ISSUE THREAD

**Issue by mika76** [yarnpkg/yarn#2724]
bug in yarn 0.20.3 (installed through chocolatey)

**What is the current behavior?**
when running [yarn help] the help is displayed but the textcolor is changed and left to a dark color, so nothing can be read afterwards.

**If the current behavior is a bug, please provide the steps to reproduce.**
Just run yarn help in powershell (I also have poshgit installed in case that makes a difference)

**What is the expected behavior?**
The text color should be reset to whatever it was before yarn help was called

**Please mention your node.js, yarn and operating system version.**
Node 7.5.0
Yarn 0.20.3

As a work around I can run `[Console]::ResetColor()` in powershell to reset the text

**Comment by Daniel15** [comment]
Hmm, I can't reproduce this issue on my computer. Can you please post your PowerShell version? Run `$PSVersionTable` and copy+paste all the output.

**Comment by mrmckeb** [comment]
This may or may not be applicable, but note that in Windows 10's upcoming update, Microsoft is supporting 256 colours (I've tested in the insider program in Bash).

**Comment by ViceIce** [comment]
Same here :(

Since upgrading nodejs from 7.3 to 7.6.
Using Windows 10 1607 (Build 14393.693).

![image](https://cloud.githubusercontent.com/assets/1798109/23255379/a4f861d6-f9bb-11e6-9bc5-ac1f6bc3f53c.png)

**Comment by ViceIce** [comment]
I can see the same behavior with grunt. So I think is has somthing to do with [libuv 1.11.0](https://github.com/libuv/libuv/pull/1143) from [Node.js 7.6.0](https://github.com/nodejs/node/blob/master/doc/changelogs/CHANGELOG_V7.md#2017-02-21-version-760-current-italoacasas).

**Comment by kumarharsh** [comment]
It's having similar (but different from each other) effects on Powershell, CMD, and Powershell-under-ConEMU:

CMD:
before running `yarn help`
![image](https://cloud.githubusercontent.com/assets/760112/24745535/a97d0660-1ad3-11e7-9275-1543cc76024f.png)

after:
![image](https://cloud.githubusercontent.com/assets/760112/24745560/b9b94f34-1ad3-11e7-9c8f-f21e30313d79.png)

Notice how the text is whiter.

Powershell:
before:
![image](https://cloud.githubusercontent.com/assets/760112/24745581/d0401eb8-1ad3-11e7-878a-0e77cf08da1b.png)

after:
![image](https://cloud.githubusercontent.com/assets/760112/24745593/d83edea6-1ad3-11e7-99c0-294e8a238ea7.png)


Conemu:
before:
![image](https://cloud.githubusercontent.com/assets/760112/24745627/fa838fc0-1ad3-11e7-877c-1f58de631e7b.png)

after:
![image](https://cloud.githubusercontent.com/assets/760112/24745628/fae9f5a8-1ad3-11e7-943e-39f8e5fbe7bf.png)

Also, in ConEMU, it messes the indentation, the yarn progress-bars don't show, the text color becomes transparent:

![image](https://cloud.githubusercontent.com/assets/760112/24745654/1251fb46-1ad4-11e7-8492-f3cd95dbeec7.png)
![image](https://cloud.githubusercontent.com/assets/760112/24745673/1b9330a8-1ad4-11e7-8c2a-744d445217bc.png)

**Comment by bestander** [comment]
So when the PR with a fix is coming? :)

**Comment by mika76** [comment]
Yeah I was using con emu as well when I first noticed this...

**Comment by prasannavl** [comment]
Yup, this is quite annoying. It not just changes the color. It messes things up. If you do a reset console, it's seemingly fine. But, any other program that uses color after that is messed up. 

A program that displays correct colors - notice the "Cyan"

![image](https://cloud.githubusercontent.com/assets/559409/25535186/d33cb216-2c53-11e7-98c5-3d25233e79fa.png)

Now, even after the console reset, running the above program again, seems to have messed up colors. 

![image](https://cloud.githubusercontent.com/assets/559409/25535251/080ecd76-2c54-11e7-96c4-4af4a289ac19.png)

**Comment by ViceIce** [comment]
This seems to be the issue: chalk/chalk#145

**Comment by ViceIce** [comment]
I would like to submit a PR to workaround this problem by simply reset colors after bold usage.
Would this be an options until ms fixed the escape codes?

Until then using legacy console mode fixes the problem too, but then all new features (like copy&paste) are disabled. :-(

**Comment by mika76** [comment]
@ViceIce I vote yes - although will this only reset the color and not indentation? I have the same problem reported by @kumarharsh in ConEmu. I guess anything is better 😄

**Comment by ViceIce** [comment]
PR should work now but failed with test timeouts. 😞

**Comment by bestander** [comment]
Is the reset only for win platform?

Tests may fail for other reasons, they can be flaky at times

**Comment by ViceIce** [comment]
Yes. I can make it conditional.

Then i have to modify the [ConsoleReporter.header](/yarnpkg/yarn/blob/25890c8cf90ff23c3230ab7b8debe73ef9a80b34/__tests__/reporters/console-reporter.js#L25) test, because the output will be conditional too.

**Comment by bestander** [comment]
Sounds good, thanks

**Comment by ViceIce** [comment]
Done.

**Comment by anfomin** [comment]
This bug still presents in 0.27.5 on my Windows 10 machine:
<img width="860" alt="2017-07-02 19 06 25" src="https://user-images.githubusercontent.com/1905503/27770540-c82dc836-5f59-11e7-9dd0-65ea389bed85.png">

**Comment by Daniel15** [comment]
@anfomin  - It's likely the fix hasn't yet been cherry-picked into a stable release. You could try the latest nightly and see if it's fixed there. https://nightly.yarnpkg.com/latest.msi

**Comment by anfomin** [comment]
@Daniel15 I've just tries version you specified. Bug is still there:

<img width="592" alt="2017-07-03 8 26 06" src="https://user-images.githubusercontent.com/1905503/27777103-56271a60-5fc9-11e7-9f10-1745596f29ea.png">

This happens only in **non-legacy Powershell** console. If I switch to legacy mode then colors are OK.

<img width="382" alt="2017-07-03 8 27 36" src="https://user-images.githubusercontent.com/1905503/27777126-858e4db4-5fc9-11e7-8fa0-f2321b904f22.png">

**Comment by Arminius** [comment]
Hi, I tried the release candidate Yarn 0.28.4 and I also get the same colour issue. This problem isn't fixed yet, can you reopen it?

**Comment by arcanis** [comment]
Can you try the [nightly builds](https://yarnpkg.com/en/docs/nightly)?

**Comment by shiftkey** [comment]
xref comment by @BYK here https://github.com/yarnpkg/yarn/issues/4529#issuecomment-331729429

> Also, this will be fixed with 1.1.0.

I'm still seeing this on Windows `10.0.15063`, yarn `1.1.0` and PowerShell `5.1.15063.632` non-legacy console, (which is the default):

<img width="873" src="https://user-images.githubusercontent.com/359239/31368800-a3643560-adcb-11e7-887e-0a9f4bbc4eee.png">

Happy to help troubleshoot further, let me know if there's anything else you need from me.

**Comment by bestander** [comment]
@shiftkey, could you check whether the fix from https://github.com/yarnpkg/yarn/pull/3607 gets executed in your environment?
Also looks like the fix from chalk has some problems and is blocked by some Windows bug https://github.com/chalk/chalk/issues/145.

@mika76 and everyone else can you confirm if the fix works for you?

**Comment by ViceIce** [comment]
Have seen this error before i upgraded to Fall Creator's update yesterday. There the unbold is fixed.

I will check this on an other pc without Fall Creator's update.

**Comment by ViceIce** [comment]
I've looked at it and it seems that the condition fails, because the `process.env.TERM` is `undefined`

**Comment by ViceIce** [comment]
PR was merged. Next nightly should have the fix. Hopefully it will get into next stable release.

**Comment by shiftkey** [comment]
@bestander apologies for the delay - `1.3.2` looks good here:

<img width="774" src="https://user-images.githubusercontent.com/359239/32410412-6583be44-c213-11e7-81aa-a1830c56248e.png">

**Comment by craigholm** [comment]
With v1.3.2 I continue to have problems with cmd.exe:

![testcolor](https://user-images.githubusercontent.com/1476786/32809910-cc6ba392-c966-11e7-9d58-e89bed2233f0.png)

**Comment by Daniel15** [comment]
Please file a separate issue  - This one was tracking PowerShell specifically.
